### DISCUSSION TITLE
Peacock segmentation fault on Mac Monterey

### DISCUSSION THREAD

**Discussion by jing-chen1122** [idaholab/moose#20114] [Q&A Getting Started] [answered]
Hi,

I am having problem when trying to run peacock on my Mac Monterey:
( ~ % mamba activate moose
~ % cd ~/projects/moose
moose % peacock
No executable found for method type(s): opt, dbg, oprof, devel
zsh: segmentation fault  peacock )

rocess:               python3.7 [46382]
Path:                  /Users/USER/*/python
Identifier:            python3.7
Version:               ???
Code Type:             X86-64 (Translated)
Parent Process:        zsh [45795]
Responsible:           Terminal [45793]
User ID:               501

Date/Time:             2022-01-24 19:04:20.6359 +0800
OS Version:            macOS 12.0.1 (21A559)
Report Version:        12
Anonymous UUID:        9E42E1B1-160E-DCEC-5CB4-4D30DB93E2F3

Sleep/Wake UUID:       B72ACBDD-7B44-42E5-8462-ADC38D2438DD

Time Awake Since Boot: 11000 seconds
Time Since Wake:       5309 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000131
Exception Codes:       0x0000000000000001, 0x0000000000000131
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [46382]

Is there any solution to this issue? Thanks in advance!

**Comment by milljm** [comment, marked as answer]
The error is indicating you must first build a binary for which Peacock can interface with. For MOOSE, this usually means either building the test suite, or modules:

```bash
cd ~/projects/moose/test
make
# or
cd ~/projects/moose/modules
make
```
Once either (or both) of those locations are built, you can then invoke peacock while somewhere within that path:

```bash
cd ~/projects/moose/test/tests/kernels/simple_diffusion
peacock -i simple_diffusion.i
#       ^instructing peacock to run the simple_diffusion.i input file in this directory
```
`peacock` is designed to search for a binary in this current directory. If not found, it goes 'up' one directory and searches there. It does this until it can't go up any more directories.
